src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -1072,14 +1119,19 @@ public int calculateMaxWaitInSeconds(int fibonacciSequenceIndex) {
         return MAX_WAIT_IN_SECONDS_IF_BDIO_UNAVAILABLE;
     }
 
+    public int getFibonacciSequenceIndex() {
```

Comment:
Would it be possible to pass in the scan mode here so that we can conditionally execute a check based on what scan type we are dealing with? I know sometimes finding a setting and passing it through is pretty ugly. Going to approve as I think this is likely fine and I don't want to hold anything up if the answer to my question is, "it's not easy", as I expect.

Reply:
Yeah that makes sense to me - by scan mode, I'm assuming you mean conditionally check if signature scan is being run? I had initially considered implementing it that way by leveraging the detectConfigurationFactory object, but I found that this object only provides me with the properties explicitly being set by the user when they run Detect. Since signature scan is also invoked by default (when we don't provide any values to `detect.tools`) and we would need to resolve additional filters like `detect.tools.excluded` if present, I figured it would be simpler to check if a scan folder exists which would only exist if signature scan was executed.

Reply:
Yeah I think you'd have to add some additional plumbing to see what the filter says should be run and then pass that information down to this area. I don't think it's worth holding up this MR for that. I'd consider this approved from my perspective.